src/js/relayclient/RelayClient.js

```diff
@@ -377,6 +386,28 @@ class RelayClient {
             }
             let relayMaxNonce = (await this.web3.eth.getTransactionCount(relayAddress)) + allowed_relay_nonce_gap;
 
+            //on first found relay, call canRelay to make sure that on-chain this request can pass
+            if ( options.validateCanRelay && firstTry ) {
+                firstTry = false;
```

Comment:
If the recipient is whitelisting relays, we can still have an error in case the first relay is whitelisted so `canRelay()` succeeds but the relay itself didn't respond, and we move on to the second relay which is not whitelisted - then we count on the relay to run it and return the correct response to client as it was before.